libsolidity/formal/BMC.cpp

```diff
@@ -863,13 +864,18 @@ void BMC::checkCondition(
 
 			for (auto const& eval: sortedModel)
 				modelMessage << "  " << eval.first << " = " << eval.second << "\n";
+
+			auto extraInfo = SecondarySourceLocation().append(modelMessage.str(), SourceLocation{});
+			auto callStack = SMTEncoder::callStackMessage(_callStack);
+			if (callStack.infos.size() > 0)
+				extraInfo = extraInfo.append(SMTEncoder::callStackMessage(_callStack));
+			if (secondaryLocation.infos.size() > 0)
+				extraInfo = extraInfo.append(move(secondaryLocation));
```

Comment:
Maybe it would be better to add the non-empty check in `SecondarySourceLocation.append()` to solve the issue more generally? Or add another method like `appendIfNotEmpty()`?

Reply:
Actually, I just noticed in a different place:
```c++
		if (call.second)
			callStackLocation.append("", call.second->location())
```
So apparently we sometimes do want the string to be empty?

Reply:
Yea, in this case I just want to print the SourceLocation. The two ifs in BMC.cpp were actually not necessary because `SecondaryLocation::append(SecondaryLocation&& _other)` simply does `infos += other.infos`, so if it's empty nothing happens anyway.